### PR TITLE
Added NTNet support to window doors

### PR DESCRIPTION
**Pull request by robbym**
[Changelogs]: # (Your PR should contain a detailed changelog of notable changes, titled and categorized appropriately. This includes, new features, sprites, sounds, balance changes, admin tools, map edits, removals, big refactors, config changes, hosting changes and important fixes. An example changelog has been provided below for you to edit. If you need additional help, read https://github.com/tgstation/tgstation/wiki/Changelogs)
Copied and modified the NTNet code from airlock.dm
I tested it with a custom door remote circuit, and it opens and closes doors I have access to (via passkey), and ignores ones I don't have access to. Also tested with a drone that paths through window doors, and it opens it successfully.

:cl: robbylm
add: Added NTNet support to window doors
fix: Fixes issue (#36859)
/:cl:

[why]: # (Please add a short description [two lines down] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.)
To fix an issue with advanced pathfinding.

### CODE REVIEW

File: code/game/machinery/doors/windowdoor.dm [hunk 1]
```diff
@@ -295,6 +299,33 @@
...
+
+	// Check packet access level.
+	if(!check_access_ntnet(data))
+		return
+
+	// Handle recieved packet.
+	var/command = lowertext(data.plaintext_data)
+	var/command_value = lowertext(data.plaintext_data_secondary)
+	switch(command)
+		if("open")
```
Comment (kevinz000) [inline code/game/machinery/doors/windowdoor.dm]: should be toggle not open

**Comment by arsserpentarium** [comment]
Thanks

File: code/game/machinery/doors/windowdoor.dm (hunk 1 above)
Comment (kevinz000) [reply code/game/machinery/doors/windowdoor.dm]: actually is this the same as airlocks?

File: code/game/machinery/doors/windowdoor.dm [hunk 2]
```diff
@@ -295,6 +299,33 @@
...
+			if(command_value == "on" && !density)
```
Comment (kevinz000) [inline code/game/machinery/doors/windowdoor.dm]: are these the same as the ones in airlock.dm?

File: code/game/machinery/doors/windowdoor.dm (hunk 1 above)
Comment (robbym) [reply code/game/machinery/doors/windowdoor.dm]: Yeah I copied it from airlocks and removed the bolt and emergency commands.

Comment (robbym) [reply code/game/machinery/doors/windowdoor.dm]: Should the airlock command be changed to "toggle" as well? Either way, I feel like it should be consistent across both door types.

Comment (robbym) [reply code/game/machinery/doors/windowdoor.dm]: Also removed AI check as it doesn't have attack_ai AFAIK.

**Comment by arsserpentarium** [comment]
In airlocks works on,off, toggle commands. So looks like this place is correct.

**Comment by robbym** [comment]
Does this require the upcoming feature thaw?

File: code/game/machinery/doors/windowdoor.dm (hunk 1 above)
Comment (PKPenguin321) [reply code/game/machinery/doors/windowdoor.dm]: Leave it as "open"

Comment (arsserpentarium) [reply code/game/machinery/doors/windowdoor.dm]: airlock have 3 commands:"bolt", "open", and "emergency".It goes to primary text field.Open/bolt is selfdescriptive.Emergency disables access check.
These commands have 3 possible values."on","off","toggle".It goes to secondary text.

also, windoors should could to be bolted,btw.

Comment (arsserpentarium) [reply code/game/machinery/doors/windowdoor.dm]: You certainly can stuck windoor open. Ingame.
But after looking at code, I can't say if it's bug or intended feature. Look at close proc.

Comment (robbym) [reply code/game/machinery/doors/windowdoor.dm]: I can see that you can emag a door. I can also see that both the open and close proc support a `forced` parameter, but cannot find a call to it except in `/obj/machinery/door/airlock/receive_signal(datum/signal/signal)`.

Comment (robbym) [reply code/game/machinery/doors/windowdoor.dm]: The `locked` and `emergency` variable is not modified in windowdoor.dm and the parent door.dm, only airlock.dm.

Comment (arsserpentarium) [reply code/game/machinery/doors/windowdoor.dm]: Well, I can lock windoor open in game. And can't find, where it is in code.

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by PKPenguin321** [comment]
reeeee

**Pull request closed without merging**